src/shared/bpf-program.c

```diff
@@ -11,6 +11,49 @@
 #include "memory-util.h"
 #include "missing_syscall.h"
 #include "path-util.h"
+#include "string-table.h"
+
+static const char *const bpf_cgroup_attach_type_table[__MAX_BPF_ATTACH_TYPE] = {
+        [BPF_CGROUP_INET_INGRESS] = "ingress",
+        [BPF_CGROUP_INET_EGRESS] = "egress",
+        [BPF_CGROUP_INET_SOCK_CREATE] = "sock_create",
+        [BPF_CGROUP_SOCK_OPS] = "sock_ops",
+        [BPF_CGROUP_DEVICE] = "device",
+        [BPF_CGROUP_INET4_BIND] = "bind4",
+        [BPF_CGROUP_INET6_BIND] = "bind6",
+        [BPF_CGROUP_INET4_CONNECT] = "connect4",
+        [BPF_CGROUP_INET6_CONNECT] = "connect6",
+        [BPF_CGROUP_INET4_POST_BIND] = "post_bind4",
+        [BPF_CGROUP_INET6_POST_BIND] = "post_bind6",
+        [BPF_CGROUP_UDP4_SENDMSG] = "sendmsg4",
+        [BPF_CGROUP_UDP6_SENDMSG] = "sendmsg6",
+        [BPF_CGROUP_SYSCTL] = "sysctl",
+        [BPF_CGROUP_UDP4_RECVMSG] = "recvmsg4",
+        [BPF_CGROUP_UDP6_RECVMSG] = "recvmsg6",
+        [BPF_CGROUP_GETSOCKOPT] = "getsockopt",
+        [BPF_CGROUP_SETSOCKOPT] = "setsockopt",
+};
+
+DEFINE_STRING_TABLE_LOOKUP(bpf_cgroup_attach_type, enum bpf_attach_type);
+
+static int bpf_program_get_info_by_fd(int prog_fd, struct bpf_prog_info *info) {
+        union bpf_attr attr;
+        uint32_t info_len;
+
+        assert(info);
+
+        info_len = sizeof(*info);
+
+        attr = (union bpf_attr) {};
```

Comment:
What I meant here was something along the lines of the usual pattern we use, and initialize directly in the declaration:

```
union bpf_attr attr {
  .info.bpf_fd = prog_fd,
  .info.info_len = info_len,
  .info.info = PTR_TO_UINT64(info),
};
```

Reply:
(PTR_TO_UINT64 is just a double cast so it's safe to use it before the assert AFAIK)